upgrade.py

```diff
@@ -174,6 +174,44 @@ def restore_file(src_base, f, d=None, attr=None):
             else:
                 logger.log("WARNING: /%s did not exist in the backup image." % f)
 
+        def convert_interface_rename_data(src, dest):
+            """
+            Convert data of interface-rename to the data consumed by xcp-networkd.
+            src - the mount point of backup partition which contains the interface-rename data.
+            dest - the mount point of the root partition of installation.
+            """
+            interface_rename_dir = os.path.join(
+                src,
+                'etc/sysconfig/network-scripts/interface-rename-data/'
+            )
+            src_file_path = os.path.join(interface_rename_dir, 'dynamic-rules.json')
+            dest_file_path = os.path.join(
+                dest,
+                constants.FIRSTBOOT_DATA_DIR,
+                'initial_network_device_rules.conf'
+            )
+            net_devs = []
+            if os.path.exists(src_file_path):
+                net_devs = netutil.net_devs_of_last_boot(src_file_path)
+            else:
+                logger.log(f"The interface-rename data file {src_file_path} does not exist.")
```

Comment:
Should the function return early here?

In future upgrades, there will never be any interface rename data so continuing to create the file below seems unnecessary.